browser/ai_chat/ai_chat_agent_profile_helper.cc

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/ai_chat_agent_profile_helper.h"
+
+#include "base/path_service.h"
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/ai_chat/core/common/buildflags/buildflags.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/constants/brave_constants.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/common/chrome_paths.h"
+
+#if !BUILDFLAG(IS_ANDROID)
+#include "chrome/browser/profiles/profile_window.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_window/public/browser_window_features.h"
+#include "chrome/browser/ui/views/side_panel/side_panel_ui.h"
+#endif
+
+static_assert(BUILDFLAG(ENABLE_BRAVE_AI_CHAT_AGENT_PROFILE));
+
+namespace ai_chat {
+
+namespace {
+
+void OpenBrowserWindowAndSidePanel(base::OnceCallback<void(Browser*)> callback,
+                                   Profile* profile) {
+#if !BUILDFLAG(IS_ANDROID)
+  // Open browser window
+  profiles::OpenBrowserWindowForProfile(
+      base::BindOnce(
+          [](base::OnceCallback<void(Browser*)> callback, Browser* browser) {
+#if !BUILDFLAG(IS_ANDROID)
```

Comment:
looks like the same check nested here? You probably want `NOTREACHED` for IS_ANDROID somewhere?

Reply:
removed the double nest